packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -187,17 +183,37 @@ export class EpochTracker {
         fromCache: boolean = false,
     ) {
         if (error.errorType === OdspErrorType.epochVersionMismatch) {
-            const err = {
-                ...error,
-                fromCache,
-                clientEpoch: this.fluidEpoch,
-                serverEpoch: epochFromResponse ?? undefined,
-                fetchType,
-            };
-            this.logger.sendErrorEvent({ eventName: "EpochVersionMismatch" }, err);
-            assert(!!this.fileEntry, "File Entry should be set to clear the cached entries!!");
-            // If the epoch mismatches, then clear all entries for such file entry from cache.
-            await this.persistedCache.removeEntries(this.fileEntry);
+            try {
+                // This will only throw if it is an epoch error.
+                this.checkForEpochErrorCore(epochFromResponse, error.errorMessage);
+            } catch (epochError) {
+                const err = {
+                    ...epochError,
+                    fromCache,
+                    clientEpoch: this.fluidEpoch,
+                    serverEpoch: epochFromResponse ?? undefined,
```

Comment:
FYI - I've added serverEpcoh to every error in throwOdspNetworkError

Reply:
ok, then I will remove it from here.